addons/binding/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVHandler.java

```diff
@@ -0,0 +1,226 @@
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.kvv.internal.DepartureResult.Departure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link KVVHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Maximilian Hess - Initial contribution
+ */
+public class KVVHandler extends BaseThingHandler {
+
+    /** the logger object */
+    private final Logger logger = LoggerFactory.getLogger(KVVHandler.class);
+
+    /** the config of the binding */
+    private KVVConfiguration config;
+
+    /** the channels */
+    private final List<ChannelUID> channels;
+
+    /** the most up to date status */
+    @Nullable
+    private DepartureResult departures;
+
+    /**
+     * Creates a new {@link KVVHandler}
+     *
+     * @param thing the {@link Thing} which is refered to.
+     */
+    public KVVHandler(Thing thing) {
+        super(thing);
+        this.channels = new LinkedList<ChannelUID>();
+
+        for (final Channel channel : thing.getChannels()) {
+            this.channels.add(channel.getUID());
+            logger.info(channel.getUID().getAsString());
+        }
+
+        for (final String key : thing.getConfiguration().getProperties().keySet()) {
+            logger.info(key + " -> " + thing.getConfiguration().get(key));
+        }
+
+        this.config = getConfigAs(KVVConfiguration.class);
+        this.departures = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.handleRefresh(channelUID);
+    }
+
+    /**
+     * Refreshes all {@link Channel Channels}
+     */
+    public void refreshAll() {
+        for (final ChannelUID channel : this.channels) {
+            this.handleRefresh(channel);
+        }
+    }
+
+    /**
+     * Refreshes a specific {@link Channel}
+     *
+     * @param channelUID the {@link ChannelUID} of the {@link Channel}
+     */
+    @SuppressWarnings("null")
+    private void handleRefresh(ChannelUID channelUID) {
+        final int id = Integer.parseInt(channelUID.getId().replaceAll("[^0-9]", ""));
+        final String type = channelUID.getId().substring(channelUID.getId().lastIndexOf('_') + 1);
+
+        logger.debug("Refresh for '" + channelUID.getId() + "'. Id is #" + id + ", type is '" + type + "'");
+
+        if (this.departures == null) {
+            logger.info("departure is null...");
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        if (this.departures.getDepartures().size() - 1 < id) {
+            logger.info("Train with #" + id + " does not exist.");
```

Comment:
This means that the number of departures returned is less than 10, right? If this is an expected situation, remove the logging (this is already reported by setting the channel state to "")